com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/scrbound.compute

```diff
@@ -1,538 +1,740 @@
+    return C;
+}
 
-    //uint vindex = groupID * NR_THREADS + threadID;
-    unsigned int g = groupID;
-    unsigned int t = threadID;
+float3x3 Invert3x3(float3x3 R)
+{
+    float3x3 C   = transpose(R); // Row to column
+    float    det = dot(C[0], cross(C[1], C[2]));
+    float3x3 adj = float3x3(cross(C[1], C[2]),
+                            cross(C[2], C[0]),
+                            cross(C[0], C[1]));
+    return rcp(det) * adj;
+}
 
-    const int subLigt = (int) (t/8);
-    const int lgtIndex = subLigt+(int) g*8;
-    const int sideIndex = (int) (t%8);
+float4x4 Homogenize3x3(float3x3 R)
+{
+    float4x4 M = float4x4(float4(R[0], 0),
+                          float4(R[1], 0),
+                          float4(R[2], 0),
+                          float4(0,0,0,1));
+    return M;
+}
 
-    const int eyeAdjustedLgtIndex = GenerateLightCullDataIndex(lgtIndex, g_iNrVisibLights, eyeIndex);
-    SFiniteLightBound lgtDat = g_data[eyeAdjustedLgtIndex];
+float4x4 PerspectiveProjection4x4(float a, float g, float n, float f)
+{
+    float b = (f + n) * rcp(f - n);    // Z in [-1, 1]
+    float c = -2 * f * n * rcp(f - n); // No Z-reversal
 
-    const float3 boxX = lgtDat.boxAxisX.xyz;
-    const float3 boxY = lgtDat.boxAxisY.xyz;
-    const float3 boxZ = -lgtDat.boxAxisZ.xyz;           // flip axis (so it points away from the light direction for a spot-light)
-    const float3 center = lgtDat.center.xyz;
-    const float radius = lgtDat.radius;
-    const float2 scaleXY = lgtDat.scaleXY;
+    return float4x4(g/a, 0, 0, 0,
+                      0, g, 0, 0,
+                      0, 0, b, c,
+                      0, 0, 1, 0);
+}
 
-    {
-        if(sideIndex<6 && lgtIndex<(int) g_iNrVisibLights)      // mask 2 out of 8 threads
-        {
-            float3 q0, q1, q2, q3;
-            GetHullQuad(q0, q1, q2, q3, boxX, boxY, boxZ, center, scaleXY, sideIndex);
+/* ------------------------------ Implementation ---------------------------- */
 
+// Improve the quality of generated code at the expense of readability.
+// Remove when the shader compiler is clever enough to perform this optimization for us.
+#define DUMB_COMPILER
 
-            const float4 vP0 = mul(g_mProjection, float4(q0, 1));
-            const float4 vP1 = mul(g_mProjection, float4(q1, 1));
-            const float4 vP2 = mul(g_mProjection, float4(q2, 1));
-            const float4 vP3 = mul(g_mProjection, float4(q3, 1));
+#ifdef SHADER_API_XBOXONE
+// The Xbox shader compiler expects the lane swizzle mask to be a compile-time constant.
+// In our case, the mask is a compile-time constant, but it is defined inside a loop
+// that is unrolled at the compile time, and the constants are generated during the
+// constant propagation pass of the optimizer. This works fine on PlayStation, but does not work
+// on Xbox. In order to avoid writing hideous code specifically for Xbox, we disable the support
+// of wave intrinsics on Xbox until the Xbox compiler is fixed.
```

Comment:
let's add a date (or xbox compiler version, so we know which version we are refering too when we read the comment later.

Reply:
OK